File: src/main/java/dev/wiisportsresorts/duelsplugin/DuelType.java
package dev.wiisportsresorts.duelsplugin;

import org.bukkit.plugin.java.JavaPlugin;

import java.util.Arrays;
import java.util.stream.Collectors;

public enum DuelType {
  SUMO, CLASSIC;

  public static void enqueue(DuelRequest request) {
    JavaPlugin.getPlugin(DuelsPlugin.class).getLogger().info(String.format("processing duel: type = %s, req = %s, " +
      "rec = %s", request.type, request.requester, request.receiver));
    switch (request.type) {
      case SUMO:
        Duels.sumo.enqueue(request);
        break;
      case CLASSIC:
        Duels.classic.enqueue(request);
        break;
    }
  }

  public static String valuesAsString() {
    return Arrays.stream(DuelType.values()).map(Enum::name).map(String::toLowerCase).collect(Collectors.joining(", "));
  }

  public static boolean isValidType(String duelType) {
    return Arrays.stream(DuelType.values()).anyMatch(type -> type.name().toLowerCase().equals(duelType.toLowerCase()));
  }
}

File: src/main/kotlin/dev/wiisportsresorts/duelsplugin/DuelType.kt
package dev.wiisportsresorts.duelsplugin

import dev.wiisportsresorts.duelsplugin.DuelsPlugin
import dev.wiisportsresorts.duelsplugin.duels.ClassicDuel
import dev.wiisportsresorts.duelsplugin.duels.NodebuffDuel
import dev.wiisportsresorts.duelsplugin.duels.SumoDuel
import org.bukkit.plugin.java.JavaPlugin
import java.util.*
import java.util.stream.Collectors

enum class DuelType {
    SUMO, CLASSIC, NODEBUFF;

    companion object {
        fun enqueue(request: DuelRequest) {
            JavaPlugin.getPlugin(DuelsPlugin::class.java).logger.info(
                "processing duel: type = ${request.type}, req = ${request.requester}, rec = ${request.receiver}"
            )
            when (request.type) {
                SUMO -> SumoDuel.enqueue(request)
                CLASSIC -> ClassicDuel.enqueue(request)
                NODEBUFF -> NodebuffDuel.enqueue(request)
            }
        }

        fun valuesAsString(): String {
            return Arrays.stream(values()).map { obj: DuelType -> obj.name }.map { obj: String -> obj.toLowerCase() }.collect(Collectors.joining(", "))
        }

        fun isValidType(duelType: String): Boolean {
            return Arrays.stream(values()).anyMatch { type: DuelType -> type.name.toLowerCase() == duelType.toLowerCase() }
        }
    }
}